### PR TITLE
Add preview tab for mobile & hide larger preview

### PR DESCRIPTION
**Pull request by shamoon**
As discussed in #129

### CODE REVIEW

**Comment by jonaswinkler** [comment]
In case the layout changes from small to larger, and the preview is being displayed, we've got two documents on the screen. It would be nice if the first tab is selected in that case. The tab nav has a directive `[(activeId)]` for that, I'm just not sure on where to listen for an event, or even if there's an event for this at all.

**Comment by shamoon** [comment]
Hmm, this might be a bit of a faux use-case, like most people aren't changing their screen size from desktop to mobile, on mobile they dont change, on desktop maybe small-is adjustments but not down to < 480px!

But anyway Im not aware of any way other than janky stuff like listening for window resizing (generally not performant) or something much more... complicated: https://medium.com/angular-in-depth/improve-performance-with-lazy-components-f3c5ff4597d2

**Comment by shamoon** [comment]
I think I misunderstood what you said, now I see the two previews, that part was easy to fix (https://github.com/jonaswinkler/paperless-ng/pull/132/commits/08495deebbd9e6a47324ab9f2ec9412b81bf333d) but switching tabs was actually pretty tough to solve! (Think I got it)

File: src-ui/src/app/components/document-detail/document-detail.component.ts [hunk 1]
```diff
@@ -48,8 +48,17 @@ export class DocumentDetailComponent implements OnInit {
     tags: new FormControl([])
   })
 
+  @ViewChild('nav') nav: NgbNav
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-detail/document-detail.component.ts:51]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/jonaswinkler/paperless-ng/pullRequest?prid=7592509)
```suggestion
  @ViewChild("nav") nav: NgbNav
```

File: src-ui/src/app/components/document-detail/document-detail.component.ts [hunk 2]
```diff
@@ -48,8 +48,17 @@ export class DocumentDetailComponent implements OnInit {
...
+  @ViewChild('pdfPreview') set pdfPreview(element) {
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-detail/document-detail.component.ts:52]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/jonaswinkler/paperless-ng/pullRequest?prid=7592509)
```suggestion
  @ViewChild("pdfPreview") set pdfPreview(element) {
```

File: src-ui/src/app/components/document-detail/document-detail.component.ts [hunk 3]
```diff
@@ -48,8 +48,17 @@ export class DocumentDetailComponent implements OnInit {
...
+    // this gets called when compontent added or removed from DOM
+    if (element && element.nativeElement.offsetParent !== null) { // its visible
+
+      setTimeout(()=> this.nav?.select(1));
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-detail/document-detail.component.ts:56]: Codacy has a fix for the issue: [Missing space before =>.](https://app.codacy.com/gh/jonaswinkler/paperless-ng/pullRequest?prid=7592509)
```suggestion
      setTimeout(() => this.nav?.select(1));
```

**Pull request closed without merging**